src/index.ts

```diff
@@ -341,6 +317,32 @@ export default <ExportedHandler<Env>>{
 						});
 					}
 					const message = await res.json<APIMessage>();
+					const [, lastModalMessage] = await Promise.all([
+						// Create a thread from this message, asking the user for more information
+						fetch(`${RouteBases.api}${Routes.threads(message.channel_id, message.id)}`, {
+							headers: {
+								Authorization: `Bot ${env.DISCORD_BOT_TOKEN}`,
+								"Content-Type": "application/json",
+							},
+							method: "POST",
+							body: JSON.stringify({
+								name: question.substring(0, 17) + "...",
+							}),
+						}),
+						getDOState(env.ChannelDO)
+					]);
+					if (lastModalMessage) {
+						await deleteMessage(
+							env.DISCORD_QUESTION_CHANNEL,
+							lastModalMessage,
+							env.DISCORD_BOT_TOKEN
+						);
+					}
+					const modalMessage = await sendModalMessage(
+						env.DISCORD_QUESTION_CHANNEL,
+						env.DISCORD_BOT_TOKEN
+					);
+					await setDOState(env.ChannelDO, (await modalMessage.json<APIMessage>()).id);
```

Comment:
Let's check if this request is successful before saving the DO state - this may throw otherwise